src/24-hour-rule.ts

```diff
@@ -34,9 +41,45 @@ export const getMinimumOpenTime = (
   return MINIMUM_MAJOR_OPEN_TIME;
 };
 
-export const shouldPRHaveLabel = (
+/**
+ *
+ * @param {Context['github']}  github An Octokit instance
+ * @param {EventPayloads.WebhookPayloadPullRequestPullRequest} pr
+ * @returns {number} a number representing the that cation should use as the
+ * open time for the PR in milliseconds, taking draft status into account.
+ */
+export const getPROpenedTime = async (
+  github: Context['octokit'],
   pr: EventPayloads.WebhookPayloadPullRequestPullRequest,
-): boolean => {
+): Promise<number> => {
+  const [owner, repo] = pr.base.repo.full_name.split('/');
+
+  // Fetch PR timeline events.
+  const { data: events } = (await github.issues.listEventsForTimeline({
+    owner,
+    repo,
+    issue_number: pr.number,
+  })) as RestEndpointMethodTypes['issues']['listEventsForTimeline']['response'];
+
+  // Filter out all except 'Ready For Review' events.
+  const readyForReviewEvents = events
+    .sort(({ created_at: cA }, { created_at: cB }) => {
+      return new Date(cB).getTime() - new Date(cA).getTime();
+    })
+    .filter(e => e.event === 'ready_for_review');
```

Comment:
nit: it's slightly more efficient to loop once and avoid sorting, since we only need the latest `ready_for_review` event.

Something like this (unchecked, untested code):
```js
const lastReadyForReviewEventTime = events.reduce((prevLatest, { event, created_at }) => {
  if (event === 'ready_for_review' && created_at > prevLatest) {
    return created;
  else {
    return prevLatest;
  }
}, new Date(0));
```

A `.reduce` is probably less readable though and depending on the number of events in general per PR this might be an over-optimization. 😅

Reply:
I did it as I did just for general readibility/straightforwardness - generally speaking there aren't many events so what would you think about returning to this if we notice a lag or perf issue?